File: examples/runfiles_fsharp/RunfilesTest.fs
module RunfilesExample

open NUnit.Framework
open System.Linq
open Bazel
open System.IO

[<TestFixture>]
type RunfilesTest () =
    [<Test>]
    member this.ShouldBeAbleToReadDataFile() =
        let runfiles = Runfiles.Create()
        let dataFilePath = runfiles.Rlocation("examples/runfiles_fsharp/data-file")
        let data = File.ReadAllLines(dataFilePath)[0]
        Assert.AreEqual("SOME CRAZY DATA!", data)

File: examples/runfiles_csharp/RunfilesTest.cs
using NUnit.Framework;
using Bazel;
using System.IO;

[TestFixture]
public sealed class RunfilesTest {
    [Test]
    public void ShouldBeAbleToReadDataFile() {
        var runfiles = Runfiles.Create();
        var dataFilePath = runfiles.Rlocation("examples/runfiles_csharp/data-file");
        var data = File.ReadAllLines(dataFilePath)[0];
        Assert.AreEqual("SOME CRAZY DATA!", data);
    }
}
